src/app/components/ebi-submission-menu/ebi-submission-menu.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'ebi-submission-menu',
+  templateUrl: './ebi-submission-menu.component.html',
+  styleUrls: ['./ebi-submission-menu.component.scss'],
+})
+export class EbiSubmissionMenuComponent implements OnInit {
+  @Input() submission: Object;
+
+  tabLinks = [
+    {'title': 'Overview', 'href': '/submission/overview'},
+    {'title': 'Project', 'href': '/submission/project'},
+    {'title': 'Contacts', 'href': '/submission/contacts'},
+    {'title': 'Data', 'href': '/submission/data'},
+    {'title': 'Submit', 'href': '/submission/submit'},
+  ];
+
+  constructor(
+    private router: Router
+  ) {
+  }
+
+
+  ngOnInit(): void {
+    //Called after the constructor, initializing input properties, and the first call to ngOnChanges.
+    //Add 'implements OnInit' to the class.
+    this.updateDataTypeLinks(this.submission);
+  }
+
+
+  /**
+   * Alter menu links and add submission DataTypes as links.
+   */
+  updateDataTypeLinks(submission) {
+    // Do nothing if dataType not yet exists in the submission.
+    try {
+      if (!submission['_links']['contents']['dataTypes']) {
```

Comment:
A proper interface would help here: `submission._links.contents.dataTypes`.

Reply:
As we agreed. this is gonna be a bit complex. i've created a new ticket for cleaning up all the application and using models for all the entities

https://trello.com/c/y3mMjMnS/47-clean-up-using-interfaces-or-models-for-different-entities